OSGeo/grass
# docs: Generate manual from markdown using mkdocs

## Goal

Build user manual from Markdown using mkdocs

![image](https://github.com/OSGeo/grass/assets/5683186/f5e82875-20dc-48fe-a302-c7f503c148ec)

## How to test

1. Install requirements, see `man/mkdocs/requirements.txt`
2. Create file `/tmp/code.lua`:
```
function CodeBlock (cb)
    return pandoc.RawBlock('markdown', '```bash\n' .. cb.text .. '\n```\n')
end
```
 3. Convert HTML files to Markdown: -> see #4620
 
```
for f in `find . -name *.html`; do echo $f; cat $f | sed 's#<div class="code"><pre>#<pre><code>#g' | sed 's#</pre></div>#</code></pre>#g' | pandoc --from=html --to=markdown -t gfm --lua-filter /tmp/code.lua > ${f%%.html}.md;done
```
4. Uncomment parts commented in 6743f1a
5. Recompile GRASS (`make distclean; ./configure ...; make`)
6. Build documentation from markdown files: `(cd man; make build-mkdocs)`
7.  Open `dist.x86_64-pc-linux-gnu/docs/mkdocs/site/index.html` in web browser

## Tasks

Critical:
- [x] clean-up `mkmarkdown.py`
- [x] `man` : integrate changes from `main`
- [ ] ~~separate makefile rules (Markdown.make, MarkdownRules.make) ?~~
- [x] make install
- [x] fix `gui/script/` compilation
- [x] fix compilation issue in `raster`
- [x] fix compilation issue in `man`
- [x] fix segfault in `--md-description`
- [x] fix code blocks
- [x] fix commit link

Important:
- [x] implement `build_md.py` and indices
- [ ] solve warnings generated by mkdocs
- [ ] add missing `meta page` comments to MD files (as YAML metadata block)
- [ ] consider to keep supporting HTML in `g.extension` for the addons
- [ ] parser_standard_options: choose formatting than table
- [ ] index: choose better formatting than table
- [ ] replace html with md in links
- [x] remove toc-related code from mkmarkdown.py and change mkdocs style to show page-related TOC
- [x] fix "## DESCRIPTION" appearance from keyword list
- [x] logo points to index.html
- [x] `site_name` to "GRASS GIS 8.5.0dev Reference Manual"
- [x] fix menu items
- [x] implement parser_standard_options
- [x] implement manual_gallery
- [x] use theme footer

Nice to have:
- [ ] implement class_graphical (https://grass.osgeo.org/grass84/manuals/class_graphical.html is empty ???)
- [ ] manual_gallery: different number of images (md vs html), fix layout (run in `man` dir: `GISBASE="/home/martin/src/grass/dist.x86_64-pc-linux-gnu" ARCH="x86_64-pc-linux-gnu" ARCH_DISTDIR="/home/martin/src/grass/dist.x86_64-pc-linux-gnu" VERSION_NUMBER=8.5.0dev VERSION_DATE=2024 python3 ./build_manual_gallery.py`) and see `dist.x86_64-pc-linux-gnu/docs/markdown/site/manual_gallery.html` and https://grass.osgeo.org/grass84/manuals/manual_gallery.html

### Tasks after merge(?)

- [ ] remove rest-related scripts
- [ ] generate man from markdown 
- [ ] remove HTML-based documentation and clean-up scripts
